library/smoothstreaming/src/main/java/com/google/android/exoplayer2/source/smoothstreaming/SsMediaSource.java

```diff
@@ -192,9 +192,9 @@ private SsMediaSource(SsManifest manifest, Uri manifestUri,
       AdaptiveMediaSourceEventListener eventListener) {
     Assertions.checkState(manifest == null || !manifest.isLive);
     this.manifest = manifest;
-    this.manifestUri = manifestUri == null ? null
-        : Util.toLowerInvariant(manifestUri.getLastPathSegment()).equals("manifest") ? manifestUri
-            : Uri.withAppendedPath(manifestUri, "Manifest");
+    String lastPathSegment = Util.toLowerInvariant(manifestUri.getLastPathSegment());
+    // Using a regexp here to allow for URLs which contain (..) as their last part, e.g. http://host/content.ism/manifest(filter=0) - see #3230
+    this.manifestUri = manifestUri == null ? null : lastPathSegment.equals("manifest") ? manifestUri : lastPathSegment.matches("manifest\\(.*\\)") ? manifestUri : Uri.withAppendedPath(manifestUri, "Manifest");
```

Comment:
It might be preferable to have a single regex that matches "manifest" and "manifest(...)". Something like `"manifest(\\(.*\\))?"` (I didn't check that's right!). Then you can get rid of the `equals("manifest")` case.

Reply:
Will fix this